src/components/type-group/type-option/TypeOption.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { generateRandomString } from '../../../utils';
+
+import styles from './TypeOption.css';
+
+class TypeOption extends Component {
+    // We are using this to greatly reduce the possibility of having duplicated ids in the dom
+    idPrefix = `type-option-${generateRandomString()}#`;
+
+    render() {
+        const { id, label, groupName, children, defaultSelected, badge: Badge, selectable } = this.props;
+        const prefixedId = `${this.idPrefix}${id}`;
+        const labelClasses = classNames(styles.label, selectable && styles.selectable);
+
+        return (
+            <div className={ styles.container }>
+                <input id={ prefixedId }
```

Comment:
You can move the input inside label. That way you won't need an id.

Reply:
Cool! 👏 Didn't know about that!